submissions/SamVal007/Memory_Pair_Game/main.js

```diff
@@ -0,0 +1,134 @@
+let similarCards = [];
+let blockedClick = false;
+const images = ['Dart_Weider', 'lyuk-skajuoker', 'R2-D2', 'c-3po',
+                'printsessa-leya-organa', 'enakin-skajuoker', 'obi-van-kenobi',
+                 'dart-sidius', 'kvaj-gon-dzhinn','joda'];
+
+const mainAttributes = {
+    container: 'container',
+    game: 'game',
+    card: 'card',
+    front: 'front_card',
+    frontImg: 'img-front_card',
+    back: 'back_card',
+    backImg: 'img-back_card',
+    flipped: 'flipped',
+    hidden: 'hidden'
+};
+
+const cleanCountArray = () => similarCards = [];
+
+const flippSelectedCards = () => {
+    setTimeout(() => {
+        document.querySelectorAll(".flipped").forEach((card) => {
+            card.classList.remove(mainAttributes.flipped);
+            cleanCountArray();
+            blockedClick = false;
+        });
+    }, 900);
+};
+
+const hideSimilarCards = () => {
+    setTimeout(() => {
+        document.querySelectorAll(".flipped").forEach((card) => {
+            card.classList.add(mainAttributes.hidden);
+            cleanCountArray();
+            blockedClick = false;
+            const counterHiddenCards = document.querySelectorAll(".hidden").length;
+            const amountOfCards = (images.length * 2);
+            console.log(amountOfCards);
+            if (counterHiddenCards == amountOfCards) {
+                victoryMessage();
+            }
+        });
+    }, 600);
+};
+
+const compareSimilarCards = () => {
+    const countCards = similarCards.length;
+    if (countCards == 2) {
+        blockedClick = true;
+        if (similarCards[0] === similarCards[1]) {
+            hideSimilarCards();
+        } else {
+            flippSelectedCards();
+        }
+    }
+};
+
+const selectCard = ({
+    target
+}) => {
+    if (!blockedClick) {
+        const selectedCard = target.closest('.card');
+        if (selectedCard && !selectedCard.classList.contains('flipped')) {
+            let alt = target.parentNode.nextSibling.firstChild.getAttribute("alt");
```

Comment:
Don't use constructions like children[0], firstElementChild, parentNode, nextSibling, etc. In such way, you rely on the order of DOM elements. So in case when you will have changed design - your code will be broken. Which is bad, obviously. Use querySelector or closest, if in event, instead.

Reply:
Didnt find better way then to do in this way:
        frontImg.alt = `${item}`;
       .....
        backImg.id = frontImg.alt;
        
        let cadName = target.getAttribute("id");

   And also avoid to use "parentNode, nextSibling, etc"